packages/openapi-ts-tests/main/test/__snapshots__/3.1.x/plugins/@pinia/colada/asClass/@pinia/colada.gen.ts

```diff
@@ -1,16 +1,17 @@
 // This file is auto-generated by @hey-api/openapi-ts
 
-import type { _JSONValue, UseMutationOptions, UseQueryOptions } from '@pinia/colada';
+import { type _JSONValue, defineQueryOptions, type UseMutationOptions } from '@pinia/colada';
 
 import { client } from '../client.gen';
+import { serializeQueryKeyValue } from '../core/queryKeySerializer.gen';
 import { BarBazService, FooBazService, type Options } from '../sdk.gen';
-import type { FooBarPostData, FooBarPostResponse, FooBarPutData, FooBarPutResponse, FooPostData, FooPostResponse, FooPutData, FooPutResponse, GetFooBarData, GetFooBarResponse, GetFooData, GetFooResponse } from '../types.gen';
+import type { FooBarPostData, FooBarPostResponse, FooBarPutData, FooBarPutResponse, FooPostData, FooPostResponse, FooPutData, FooPutResponse, GetFooBarData, GetFooData } from '../types.gen';
 
 export type QueryKey<TOptions extends Options> = [
-    Pick<TOptions, 'body' | 'path'> & {
+    Pick<TOptions, 'baseUrl' | 'body' | 'path' | 'query'> & {
```

Comment:
@brolnickij why add items to the pick if we're defining them in the intersection? Shouldn't this be...

```suggestion
    Pick<TOptions, 'path'> & {
```

...instead?

Reply:
fixed!)

- https://github.com/hey-api/openapi-ts/pull/2610/commits/5ed19a928c00ee5d85022b2d82a7b602b31485bd
- https://github.com/hey-api/openapi-ts/pull/2610/commits/088c1fe65dbe7b6f5b46f36f9b39bed0485c453d